### PR TITLE
AbDisc: Sample to Plate Nav

### PR DESCRIPTION
**Pull request by RosalineP**
#### Rationale
See spec [here](https://docs.google.com/document/d/1x2Dh_hZwwTMoSdwE0HI08jytTwfo4PuT-np-Y2Lrd7M/edit?usp=sharing).

#### Related Pull Requests
- https://github.com/LabKey/labkey-ui-components/pull/1670
- https://github.com/LabKey/platform/pull/6166
- https://github.com/LabKey/labkey-ui-premium/pull/624
- https://github.com/LabKey/limsModules/pull/1020

#### Changes
- Add isPlated col

### CODE REVIEW

File: experiment/src/org/labkey/experiment/api/ExpMaterialTableImpl.java [hunk 1]
```diff
@@ -769,6 +769,38 @@ public void addQueryFieldKeys(Set<FieldKey> keys)
...
+        if (plateUserSchema != null)
+        {
+            SQLFragment existsSubquery = new SQLFragment()
+                    .append("SELECT 1 FROM ")
+                    .append(plateUserSchema.getTable("Well"), "well")
+                    .append(" WHERE well.sampleid = ").append(rowIdField);
+
+            sql = new SQLFragment()
+                    .append("CASE WHEN EXISTS (")
+                    .append(existsSubquery)
```
Comment (RosalineP) [inline experiment/src/org/labkey/experiment/api/ExpMaterialTableImpl.java:784]: Might be weird doing this 'existsSubquery' thing in the middle—readable to me, but maybe not for others? Happy to move things around.

File: assay/src/org/labkey/assay/PlateController.java [hunk 2]
```diff
@@ -1784,4 +1784,16 @@ public Object execute(ReformatOptions options, BindException errors) throws Exce
             return null;
         }
     }
+
+    // Dummy action that is used in ExpMaterialTable's IsPlated column's URL. The resulting URL is mapped via
```
Comment (labkey-nicka) [inline assay/src/org/labkey/assay/PlateController.java]: Do we need this? I would say just do not declare this action and put the comment on the DetailsURL declaration in column construction.

File: experiment/src/org/labkey/experiment/api/ExpMaterialTableImpl.java [hunk 3]
```diff
@@ -769,6 +769,38 @@ public void addQueryFieldKeys(Set<FieldKey> keys)
         if (InventoryService.get() != null && (st == null || !st.isMedia()))
             defaultCols.addAll(InventoryService.get().addInventoryStatusColumns(st == null ? null : st.getMetricUnit(), this, getContainer(), _userSchema.getUser()));
 
+        UserSchema plateUserSchema = QueryService.get().getUserSchema(_userSchema.getUser(), getContainer(), "plate");
+        String rowIdField = ExprColumn.STR_TABLE_ALIAS + "." + Column.RowId.name();
```
Comment (labkey-nicka) [inline experiment/src/org/labkey/experiment/api/ExpMaterialTableImpl.java]: `rowIdField` can be moved inside the `if` statement.

File: experiment/src/org/labkey/experiment/api/ExpMaterialTableImpl.java [hunk 4]
```diff
@@ -769,6 +769,38 @@ public void addQueryFieldKeys(Set<FieldKey> keys)
...
+        }
+        else
+        {
+            sql = new SQLFragment("SELECT NULL");
+        }
+        var col = new ExprColumn(this, Column.IsPlated.name(), sql, JdbcType.VARCHAR);
+        col.setDescription("Whether the sample that has been plated, if plating is supported.");
+        col.setUserEditable(false);
+        col.setReadOnly(true);
+        col.setHidden(true);
```
Comment (RosalineP) [inline experiment/src/org/labkey/experiment/api/ExpMaterialTableImpl.java]: This actually shouldn't be 'hidden', slight language overlap confusion from spec. Will remove this line.

**Review by labkey-nicka — APPROVED** [review]

File: experiment/src/org/labkey/experiment/api/ExpMaterialTableImpl.java (hunk 1 above)
Comment (labkey-nicka) [reply experiment/src/org/labkey/experiment/api/ExpMaterialTableImpl.java:784]: Seems fine to me.

**Pull request merged**